src/VecSim/vec_sim_tiered_index.h

```diff
@@ -158,12 +141,13 @@ VecSimTieredIndex<DataType, DistType>::topKQuery(const void *queryBlob, size_t k
 
         // Simply query the main index and return the results while holding the lock.
         this->mainIndexGuard.lock_shared();
-        auto res = this->backendIndex->topKQuery(queryBlob, k, queryParams);
+        auto res = this->backendIndex->topKQueryWrapper(queryBlob, k, queryParams);
```

Comment:
if we reached the `topKQuery` function of `VecSimTieredIndex`, doesn't it mean we already processed the blob when `topKQueryWrapper` was called?

Reply:
The frontend computer processed it. The backend computer might have additional processing to do